summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/FutureQueue.scala

```diff
@@ -141,6 +188,11 @@ private[summingbird] class FutureQueue[S, T](
     }
   }
 
+  /**
+   * Retrieve any completed results, along with their associated states.
+   *
+   * Returns up to maxEmitPerExec items.
+   */
   def dequeue: TraversableOnce[(S, Try[T])] = {
```

Comment:
just a question, can't this be `Iterable[(S, Try[T])]`? `.take` returns something iterable, right? This seems to limit the caller to only go through it once, but in fact, we materialize it into memory already.